File: Leetcode Daily Challenge/032-February-2026/006-minimum-removals-to-balance-array.java
// https://leetcode.com/problems/minimum-removals-to-balance-array/




import java.util.*;

class Solution {
    public int minRemoval(int[] nums, int k) {
        int n = nums.length;
        Arrays.sort(nums);
        int largestBalancedLen = 0;
        int i = 0, j = 0;
        while (j < n) {
            while (i < j && (long)nums[i] * k < nums[j]) i ++;
            largestBalancedLen = Math.max(largestBalancedLen, j - i + 1);
            j ++;
        }
        return (n - largestBalancedLen);
    }
}

File: Leetcode Daily Challenge/032-February-2026/006-minimum-removals-to-balance-array.py
# https://leetcode.com/problems/minimum-removals-to-balance-array/




from typing import List

class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        n = len(nums)
        nums.sort()
        i = 0
        j = 0
        largestBalancedLen = 0
        while j < n:
            while i < j and nums[i] * k < nums[j]:
                i += 1
            largestBalancedLen = max(largestBalancedLen, j - i + 1)
            j += 1
            
        return (n - largestBalancedLen)